### PR TITLE
Move to v1 of the backend API

### LINKED ISSUE #3756: Implement new `/v1` API version for version 5 settings format [5d]
We need to bump our API version to `/v1` because the `settings` API will be changed in an incompatible way.

All `/v0` endpoints should now return an error. We don't want implement the `GET` method for `/v0/settings` (`PUT` alone is simple because we need migration code for updates anyways).

The PUT method for the `/v1/settings` endpoint will return an error when the call attempts to modify any structure that (directly, or via its parents) has a `locked` attribute that is set to `true` in the current settings.

I.e. you cannot modify `containerEngine.allowedImages.enabled` if `containerEngine.allowedImages.locked` is currently `true`.

We require that every `PUT` request of the `/v1/settings` endpoint includes the `version: 5` field in the request body. That way we could make further changes to the settings layout in the future without having to bump the API version every time.

# Acceptance criteria

* All `/v0` endpoints return an error - run `rdctl api /v0/settings` should bring an error
* The `/v1/settings` endpoint supports the new v5 schema for both `GET` and `PUT`
* The `PUT` method for `/v1/settings` returns an error unless the `version` field exists and has the value `5`.

  ```
  rdctl api -X PUT -b '{"version":5,"containerEngine":{"imageAllowList":{"enabled":true,"patterns":["a","b"]}}}' /v1/settings
  ```

# Release Notes:

### PR DESCRIPTION
**Pull request by ericpromislow**
Fixes #3756

All /v0 endpoints are now errors (except for the `/v0` endpoint which just returns an array containing itself).

Signed-off-by: Eric Promislow <epromislow@suse.com>

### CODE REVIEW

**Review by mook-as — DISMISSED** [review]
I think we should land #3861 first, or at least do _something_ so people don't assume we intend to support v1, but 🤷

**Comment by ericpromislow** [comment]
I assigned that other issue the `Next` milestone, and would rather get this merged when ready. Because a bunch of other PRs depend on this PR.

**Comment by jandubois** [comment]
> I think we should land #3861 first, or at least do _something_ so people don't assume we intend to support v1, but 🤷

#3927 should be included in this release to inform users that the v1 API is still considered internal and experimental.

@mook-as Does this address your concerns? Feel free to update #3927 if you think additional notices are required.

**Comment by mook-as** [comment]
FWIW, I preferred fixing #3927 _first_ so that we keep main always shippable (if this lands first, we should make sure that gets fixed before we ship).  That includes getting info into the hands of people (expected count: 0) who randomly pull `main` and play with it.

But that's certainly a preference that's not worth frustrating people too much with.

**Comment by ericpromislow** [comment]
I think we should get this in first and write up any policy later.

**Review by mook-as — CHANGES_REQUESTED** [review]
Basically looks fine. Checked with `git grep -w v0 -- . ':!**/go.mod' ':!**/go.sum'`.

File: e2e/rdctl.e2e.spec.ts [hunk 1]
```diff
@@ -312,18 +312,18 @@ test.describe('Command server', () => {
   });
 
   test('should not restart on unrelated changes', async() => {
-    let resp = await doRequest('/v0/settings');
+    let resp = await doRequest('/v1/settings');
 
     expect(resp.ok).toBeTruthy();
-    const telemetry = (await resp.json() as Settings).application.telemetry.enabled;
+    const telemetry = (await resp.json()).application.telemetry.enabled;
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply e2e/rdctl.e2e.spec.ts]: done

**Pull request merged**